src/components/NewCommentForm.tsx

```diff
@@ -75,25 +173,50 @@ export const NewCommentForm: React.FC = () => {
             id="comment-body"
             name="body"
             placeholder="Type comment here"
-            className="textarea is-danger"
+            className={textDangerClass(textError)}
+            value={text}
+            onChange={(e) => {
+              setText(e.target.value);
+              setTextError(false);
+            }}
           />
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Enter some text
-        </p>
+        {textError && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Enter some text
+          </p>
+        )}
       </div>
 
       <div className="field is-grouped">
         <div className="control">
-          <button type="submit" className="button is-link is-loading">
+          <button
+            type="submit"
+            className={buttonClass(isLoading)}
+            onClick={(event) => {
+              event.preventDefault();
+              addComent();
+            }}
+          >
             Add
           </button>
         </div>
 
         <div className="control">
           {/* eslint-disable-next-line react/button-has-type */}
-          <button type="reset" className="button is-link is-light">
+          <button
+            type="reset"
+            className="button is-link is-light"
+            onClick={() => {
```

Comment:
```suggestion
            onClick={resetForm}
```
Consider moving this function from jsx to a separate function